### DISCUSSION TITLE
A thank you to the developer and results from 3 different MacBooks

### DISCUSSION THREAD

**Discussion by mikeatlarge** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#354] [General]
I just bought a lifetime Pro license as this app works well and should easily pay for itself in prolonged battery life. It works flawlessly on an A2337 2021 M1 MacBook Air. It also works reasonably well on an A1466 early 2015 MacBook Air and an A1398 mid 2015 MacBook Pro 15. 

On both older MacBooks the MagSafe charging connector stays orange all the time as the battery is not being fully charged. But I have confirmed charging indeed stops at whatever percentage AlDente is configured for. I'm not yet sure about altered battery calibrations but so far no issues have resulted on any of the three MacBooks. 

It's really nice to just be able to leave my MacBooks plugged in most of the time knowing they're not sitting at 100% charge and being able to save on battery cycles by not running them from the battery when there is power nearby. Both the 2015 MacBooks have their original batteries that both still have about 80 - 85% of their original capacity due to relatively careful use over the last 6 years. Had I had AlDente back in 2015 I'm sure the batteries would be in even better shape but AlDente will still help extend their life going forward.

Thank you David for solving a problem that nobody else, including Apple with their "Optimize Charging" feature that barely works, has solved as well as you have. As I mentioned in another post I favor Dell laptops for Linux and Windows as they build similar functionality right into the BIOS (so it works independently of any operating system). Dell offers both manual thresholds for where charging starts and ends or an adaptive mode that monitors your usage. It's too bad Apple doesn't give a you a similar way to natively control their battery management. 

As long as the battery lasts through Apple Care+ I suspect Apple welcomes the revenue from battery replacement and people buying a new MacBook because the battery is weak in their old one. That might explain why even their new Optimize feature works so poorly and they don't give you more direct control over charging.

Now we just need a similar app for the iPhone, iPad, and Apple Watch! The only charging apps I'm aware of for those 
devices merely generate a notification and/or sound when a certain charge level is reached. So if you're not around to hear it, or sleeping, you're out of luck.

**Comment by theGitMichel** [comment]
If you have homekit active you can use the shortcut app in combination with an eve energy or others. Simply create a shortcut: if battery exceeds x % then turn off eve energy. In most cases my battery will be charched up to 81 % and then it stopped. For me it works fine. But I didn't recognized any improvements by charing battery between 20/30-80 instead of charging it to 100 % to mostly the same time at night.
